cartographer/io/internal/mapping_state_serialization.h

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright 2018 The Cartographer Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#ifndef CARTOGRAPHER_IO_INTERNAL_MAPPING_STATE_SERIALIZATION_H_
+#define CARTOGRAPHER_IO_INTERNAL_MAPPING_STATE_SERIALIZATION_H_
+
+#include "cartographer/io/proto_stream_interface.h"
+#include "cartographer/mapping/pose_graph.h"
+#include "cartographer/mapping/proto/trajectory_builder_options.pb.h"
+
+// Helper function to serialize a PoseGraph to a protobuf stream.
+namespace cartographer {
+namespace io {
+
+// The current serialization format version.
+static constexpr int kMappingStateSerializationFormatVersion = 1;
```

Comment:
will the version number be used outside of `mapping_state_serialization.cc`?

Reply:
In a different branch I am using it to verify the version in the deserialization. We can also do it differently.

Reply:
LGTM/WLDDTM (wubba lubba dub dub to me).